synapse/handlers/sync.py

```diff
@@ -562,8 +562,21 @@ def compute_summary(self, room_id, sync_config, batch, state, now_token):
             details.get(Membership.INVITE, empty_ms).count
         )
 
-        if name_id or canonical_alias_id:
-            defer.returnValue(summary)
+        # if the room has a (non-deleted) name or canonical_alias, we can skip
```

Comment:
What do you mean by non-deleted? You can't delete state.

Reply:
well, people 'delete' it by blatting {} over it.

Reply:
clarified